mlibrary/search
# Class component conversions part 2

<details open>
<summary> :hand:&nbsp;&nbsp;<strong>Wait!</strong></summary>

Let's make sure your pull request is ready. Ask yourself...

:white_check_mark:&nbsp;&nbsp;"Did I do a good job testing my own code?"
:white_check_mark:&nbsp;&nbsp;"Is my code easy to read and understand?"
:white_check_mark:&nbsp;&nbsp;"Have I written unit tests? Do the unit tests fully cover my code?"
:white_check_mark:&nbsp;&nbsp;"Have I tested the repository's behavior to make sure my code didn't break anything?"
:negative_squared_cross_mark:&nbsp;&nbsp;"Did I include unrelated work that could be in its own pull request?"

Once you are ready, be sure to delete this `details` section before opening your pull request.
</details>

# Overview
Talk about the intent and scope of the original issue. Go over the general changes that have been made between this branch and the branch the pull request will merge into. Displaying screenshots can help!

Add links to any documentation or issues that can help give more background information. Is the pull request related to a GitHub or JIRA issue? Link to them while using [closing keywords](https://docs.github.com/articles/closing-issues-using-keywords), like so:

> This pull request closes/resolves/fixes [LIBSEARCH-1234](https://mlit.atlassian.net/browse/LIBSEARCH-1234).

## Anything else?
_Optional_. If appropriate, provide additional notes the reviewers should be made aware of. They can be about any "gotchas" or even changes that are expected to be made in the future, but are not applied to this pull request.

To align with [current React practices](https://react.dev/learn/your-first-component), these class components have been converted to functional components:
- `ComponentName`

## Testing
- Make sure the PR is consistent in these browsers:
- Run accessibility tests:
- _How to use the feature_.
